ui/panels-plugin/src/plugins/gauge-chart/GaugeChartOptionsEditor.tsx

```diff
@@ -11,13 +11,44 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+import { useEffect, useRef, useState } from 'react';
+import { TextField } from '@mui/material';
 import { OptionsEditorProps } from '@perses-dev/plugin-system';
-import { Box } from '@mui/material';
 import { GaugeChartOptions } from './gauge-chart-model';
 
 export type GaugeChartOptionsEditorProps = OptionsEditorProps<GaugeChartOptions>;
 
 export function GaugeChartOptionsEditor(props: GaugeChartOptionsEditorProps) {
-  const { value } = props;
-  return <Box>{JSON.stringify(value)}</Box>;
+  const ref = useRef<HTMLInputElement>(null);
+  const [value, setValue] = useState(JSON.stringify(props.value, null, 2));
```

Comment:
Might want to change this to the callback style to init the state so we're not running `JSON.stringify` on every render cycle.